crates/matrix-sdk/src/room/mod.rs

```diff
@@ -400,11 +399,17 @@ impl Room {
 
         self.client.join_room_by_id(self.room_id()).await?;
 
-        #[cfg(all(feature = "experimental-share-history-on-invite", feature = "e2e-encryption"))]
-        if let Some(inviter) = inviter {
-            shared_room_history::maybe_accept_key_bundle(self, inviter.user_id()).await?;
+        #[cfg(feature = "e2e-encryption")]
+        if self.client.inner.enable_share_history_on_invite {
+            if let Some(inviter) = inviter {
+                shared_room_history::maybe_accept_key_bundle(self, inviter.user_id()).await?;
+            }
         }
 
+        #[cfg(not(feature = "e2e-encryption"))]
+        // Suppress "unused variable" lint
+        let _inviter = inviter;
```

Comment:
I don't know if there is a less awful way to do this. I wanted to keep the code flow above consistent between `feature = "e2e-encryption"` and not, so didn't really want another `#[cfg(feature = "e2e-encryption")]` where `inviter` is calculated.